### PR TITLE
Ensure Windows build w/o port/port.h in public headers

### PR DESCRIPTION
**Pull request by yuslepukhin**
- Remove make file defines from public headers and use _WIN32 because it is compiler defined
 - use __GNUC__ and __clang__ to guard non-portable attributes
 - add #include "port/port.h" to some new .cc files.
 - minor changes in CMakeLists to reflect recent changes

### CODE REVIEW

**Comment by yuslepukhin** [comment]
Sorry for loosing the comments in the previous PR. Git would not co-operate.

File: CMakeLists.txt [hunk 1]
```diff
@@ -202,7 +203,6 @@ set(SOURCES
         util/xxhash.cc
         utilities/backupable/backupable_db.cc
         utilities/checkpoint/checkpoint.cc
-        utilities/compacted_db/compacted_db_impl.cc
```
Comment (siying) [inline CMakeLists.txt]: You need to sync and rebase the codes for those changes.

File: table/adaptive_table_factory.cc [hunk 2]
```diff
@@ -6,6 +6,9 @@
 #include "table/adaptive_table_factory.h"
 
 #include "table/format.h"
+#include "port/port.h"
+
+#include "port/port.h"
```
Comment (siying) [inline table/adaptive_table_factory.cc]: Duplicated?

**Comment by siying** [comment]
Great! Please rebase the codes and I think it will be ready to go.

File: CMakeLists.txt (hunk 1 above)
Comment (yuslepukhin) [inline CMakeLists.txt]: @siying I believe this is the result of it.

File: table/adaptive_table_factory.cc (hunk 2 above)
Comment (yuslepukhin) [inline table/adaptive_table_factory.cc]: @siying seems like it

File: CMakeLists.txt (hunk 1 above)
Comment (siying) [inline CMakeLists.txt]: Oh thanks.

**Comment by siying** [comment]
Maybe as a separate diff but incluce/rocksdb/c.h has yet another OS_WIN to remove.

**Comment by yuslepukhin** [comment]
@siying I think this might be the right place to do it.

**Comment by siying** [comment]
@yuslepukhin should replace it by _WIN32?

**Comment by yuslepukhin** [comment]
@siying Yes, that is the way to do currently.

**Comment by siying** [comment]
Merged. Thank you for your contribution and patient in addressing the comments.

**Pull request merged**